api/envoy/config/trace/v3/opentelemetry.proto

```diff
@@ -25,4 +38,8 @@ message OpenTelemetryConfig {
   // The name for the service. This will be populated in the ResourceSpan Resource attributes.
   // If it is not provided, it will default to "unknown_service:envoy".
   string service_name = 2;
+
+  // An ordered list of resource detectors
+  // [#extension-category: envoy.tracers.opentelemetry.resource_detectors]
+  repeated ResourceDetector resource_detectors = 3;
```

Comment:
I saw that we have [TypedExtensionConfig](https://github.com/envoyproxy/envoy/blob/v1.27.0/api/envoy/config/core/v3/extension.proto) for this purpose and I wanted to use it. The problem here is that the field `typed_config` is required. for resource detectors, it's not always that it will need its own extra config, which is the case of the environment detector in this PR. 

To circumvent this, I created this new type `ResourceDetector` which is basically the same, but does not require the `typed_config`. @htuch @KBaichoo tagging you since you commented on the issue.

Reply:
I'm working on a next detector for Dynatrace, and that one will have its own configuration.

Reply:
Ah neat, that make thing much simpler I think. I will adapt this way, thank you!

Reply:
Done in 277d9ce35c173dc2340e5ff452593b8cefd0250e. Simplified the code a bit :)